internal/app/app.go

```diff
@@ -37,65 +44,56 @@ type option func(*options)
 
 // App is the main application object.
 type App struct {
-	conn    *dbus.Conn
-	obj     *busObject
-	timeout time.Duration
+	busObject *proxyManagerBus
 }
 
 // Apply is a function called via D-Bus to apply the system proxy settings.
-func (b *busObject) Apply(http, https, ftp, socks, no, mode string) *dbus.Error {
+func (b *proxyManagerBus) Apply(http, https, ftp, socks, no, mode string) *dbus.Error {
 	// Methods calls spin up separate goroutines, so ensure we don't run them in parallel
 	b.mu.Lock()
 	defer b.mu.Unlock()
 
+	b.running = true
```

Comment:
Even if the timeout helps not triggering the race, I would add a mutex around it, wdyt?